codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -186,6 +246,19 @@ public boolean visit(Http2Stream stream) throws Http2Exception {
         if (delta > 0) {
             // The window size increased, send any pending frames for all streams.
             writePendingBytes();
+        } else if (delta != 0 && listener != null) {
+            // If the write operation does not occur we need to ensure we need to ensure listener is notified.
+            connection.forEachActiveStream(new Http2StreamVisitor() {
```

Comment:
per instance of flow controller - yes. If you want to go for static you can also make
```
    private AbstractState state(Http2Stream stream) {
```
static and it should be fine